lms/models/jwt_oauth2_token.py

```diff
@@ -0,0 +1,52 @@
+import sqlalchemy as sa
+
+from lms.db import BASE
+
+
+class JWTOAuth2Token(BASE):
+    """
+    JWT based OAuth2Token.
+
+    https://datatracker.ietf.org/doc/html/rfc7523
+
+    Similar to models.OAuth2Token but using the JWT profile required by LTIA APIs.
+
+    These tokens are not valid for a particular user but for one tool from the point of view of the LMS.
+    In our system that's identified by "LTIRegistration".
+
+    Tokens are also valid for one set of scopes only.
+    """
+
+    __tablename__ = "jwt_oauth2_token"
+    __table_args__ = (sa.UniqueConstraint("lti_registration_id", "scopes"),)
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    # The LTIRegistration that this token belongs to foreign key
+    lti_registration_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("lti_registration.id", ondelete="cascade"),
+        nullable=False,
+    )
+
+    # The LTIRegistration that this token belongs to foreign key
+    lti_registration = sa.orm.relationship(
+        "LTIRegistration", foreign_keys=[lti_registration_id]
+    )
+
+    # Scopes this token is valid for
+    scopes = sa.Column(sa.UnicodeText(), nullable=False)
+
+    # The OAuth 2.0 access token, as received from the authorization server.
+    access_token = sa.Column(sa.UnicodeText(), nullable=False)
+
+    # Time at which the toke will expire
+    expires_at = received_at = sa.Column(sa.DateTime, nullable=False)
+
+    # The time when we received `access_token` and `expires_in`.
+    received_at = sa.Column(
+        sa.DateTime,
+        default=sa.func.now(),
+        server_default=sa.func.now(),
+        nullable=False,
+    )
```

Comment:
Are we redefining the `received_at` column here? Also is this different from a `created` field?

Reply:
There's no created/updated here, just this `received_at`

Reply:
I initially used the CreatedUpdatd mixin and found it confusing to rely on it.

But now that I think about it, "expired_at" is now a date and calculated in python so we could include the mixin just for the "metadata" value.

We don't use received_at

Reply:
Now expiration is handled by `expires_at` and we have created_at and updated_at as metadata.